Source/WebKit/UIProcess/FindTextMatchesCallbackAggregator.cpp

```diff
@@ -37,12 +38,35 @@ Ref<FindTextMatchCallbackAggregator> FindTextMatchCallbackAggregator::create(Com
 
 void FindTextMatchCallbackAggregator::foundMatches(Vector<WebFoundTextRange>&& range)
 {
-    // FIXME: matches will be returned in amy order from frames. Matches need to be sorted in the order they appear in the frame tree.
     m_range.appendVector(range);
 }
 
 FindTextMatchCallbackAggregator::~FindTextMatchCallbackAggregator()
 {
+    std::ranges::sort(m_range, [](const WebFoundTextRange& a, const WebFoundTextRange& b) {
+        if (a.order != b.order)
+            return a.order < b.order;
+
+        bool aIsPDF = std::holds_alternative<WebFoundTextRange::PDFData>(a.data);
+        bool bIsPDF = std::holds_alternative<WebFoundTextRange::PDFData>(b.data);
+
+        // DOM always comes before PDF when in same frame
```

Comment:
We don't support embedded PDFs at all on iOS (only main frame), so DOM and PDF content would never be in the same frame. (So the logic is still fine, but the comment is inaccurate)